G/GSL/GSL@2/build_tarballs.jl

```diff
@@ -3,17 +3,26 @@
 using BinaryBuilder
 
 name = "GSL"
-version = v"2.7.1" # <--- This version number is a lie to build for experimental platforms
+version_string = "2.7.1"
+version = v"2.7.2" # <--- This version number is a lie to keep it different from our
+                   # previous fake "2.7.1" build, as they have different ABI because of
+                   # https://git.savannah.gnu.org/cgit/gsl.git/commit/configure.ac?id=77e7c7d008707dace56626020eaa6181912e9841
 
 # Collection of sources required to build GSL
 sources = [
-    ArchiveSource("http://ftp.gnu.org/gnu/gsl/gsl-$(version.major).$(version.minor).tar.gz",
-                  "efbbf3785da0e53038be7907500628b466152dbc3c173a87de1b5eba2e23602b"),
+    ArchiveSource("http://ftp.gnu.org/gnu/gsl/gsl-$(version_string).tar.gz",
+                  "dcb0fbd43048832b757ff9942691a8dd70026d5da0ff85601e52687f6deeb34b"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir/gsl-*/
+if [[ "${target}" == aarch64-apple-darwin* ]]; then
+    # aclocal.m4 has some lines where it expects `MACOSX_DEPLOYMENT_TARGET` to be up to
+    # version 10.  Let's pretend to be 10.16, as many tools do to make old build systems
+    # happy.
+    export MACOSX_DEPLOYMENT_TARGET="10.16"
+fi
```

Comment:
This also explains why the previous build worked: before https://github.com/JuliaPackaging/BinaryBuilderBase.jl/pull/217 we used to unconditionally set `MACOSX_DEPLOYMENT_TARGET="10.8"`.